main.go

```diff
@@ -169,18 +170,42 @@ func binaries(filter map[string]matched) ([]string, error) {
 			if strings.HasPrefix(fi.Name(), ".") {
 				continue
 			}
+			// Ignore non-exe files on Windows.
+			if fi.Name() != binaryName(canonicalName(fi.Name())) {
+				continue
+			}
 
 			// If user specified a list of binaries, filter out binaries that don't match.
 			if len(filter) != 0 {
-				if _, ok := filter[fi.Name()]; !ok {
+				if _, ok := filter[canonicalName(fi.Name())]; !ok {
 					continue
 				}
-				filter[fi.Name()] = matched(true)
+				filter[canonicalName(fi.Name())] = matched(true)
 			}
 
-			binaries = append(binaries, fi.Name())
+			binaries = append(binaries, canonicalName(fi.Name()))
 		}
 	}
 
 	return binaries, nil
 }
+
+// canonicalName, when called on a Windows system, trims the ".exe" suffix from
+// the end of a binary's filename.
+func canonicalName(anyName string) string {
+	if "windows" == runtime.GOOS {
+		return strings.TrimSuffix(anyName, ".exe")
+	} else {
+		return anyName
+	}
+}
+
+// binaryName, when called on a Windows system, adds a ".exe" suffix to a
+// basename to produce the name of a binary in the Windows filesystem.
+func binaryName(canonicalName string) string {
```

Comment:
Rename `canonicalName` to `commandName`. It's the name of the Go command (a package with name "main"), so `commandName` is a better description.